src/Cafe/IOSU/legacy/iosu_fpd.cpp

```diff
@@ -165,9 +165,32 @@ namespace iosu
 				g_fpd.myPresence.isOnline = 1;
 				g_fpd.myPresence.gameKey.titleId = CafeSystem::GetForegroundTitleId();
 				g_fpd.myPresence.gameKey.ukn = CafeSystem::GetForegroundTitleVersion();
+
+				// Resolve potential domain to IP address
+				struct addrinfo hints = {0}, *addrs;
+				hints.ai_family = AF_UNSPEC;
+				hints.ai_socktype = SOCK_DGRAM;
+				hints.ai_protocol = IPPROTO_UDP;
+
+				const int status = getaddrinfo(nexTokenResult.nexToken.host, NULL, &hints, &addrs);
+				if (status != 0) {
+					cemuLog_log(LogType::Force, "IOSU_FPD: Failed to resolve hostname {}, {}", nexTokenResult.nexToken.host, gai_strerror(status));
+					return;
+				}
+
+				if (addrs->ai_family != AF_INET) {
+					cemuLog_log(LogType::Force, "IOSU_FPD: Resolved IP for hostname {} not IPv4", nexTokenResult.nexToken.host);
+					return;
+				}
```

Comment:
If only IPv4 addresses are acceptable you should probably set it in the hints field. Are IPv6 address unsupported?

Reply:
> Are IPv6 address unsupported?

Real hardware only accepts IPv4 addresses, from my testing. I have not delved into Cemu's source enough to know if it accepts IPv6, but I know that the real servers only ever returned IPv4 addresses so I assume Cemu is built in a similar way